Lecture4/Task4/VoucherFactory.java

```diff
@@ -0,0 +1,15 @@
+import predefinedExceptions.RegexException;
+
+class VoucherFactory {
+    static Voucher getVoucher(String[] voucherDetails) throws RegexException {
+        switch (Integer.parseInt(voucherDetails[0])) {
+            case 1:
+                return new RestVoucher(voucherDetails);
+            case 2:
+                return new CureVoucher(voucherDetails);
+            case 3:
+                return new ShoppingVoucher(voucherDetails);
+        }
+        return null;
```

Comment:
don't return null. Instead define default behavior to throw custom exception if type of voucher is not recognized.

Reply:
fixed in Tasks1-4-fixed package